src/appleseed/renderer/kernel/lighting/lightsamplerbase.h

```diff
@@ -93,12 +103,14 @@ class LightSamplerBase
     
     EmitterCDF                              m_non_physical_lights_cdf;
     EmitterCDF                              m_emitting_triangles_cdf;
-
+ 
     EmittingTriangleKeyHasher               m_triangle_key_hasher;
     EmittingTriangleHashTable               m_emitting_triangle_hash_table;
-
+ 
     bool                                    m_use_light_tree;
 
+    LightHandlingLambda                     m_cdf_light_handling;
```

Comment:
I'm not sure why we need to store the lambda here?

Edit: Ah, I see, you're trying to factorize the lambda. Not a good idea, just duplicate the lambda code in both samplers.